File: app/src/main/java/com/vgaw/scaffolddemo/page/demo/compoment/other/OtherFrag.java
package com.vgaw.scaffolddemo.page.demo.compoment.other;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.widget.NestedScrollView;

import com.vgaw.scaffold.page.BaseFrag;
import com.vgaw.scaffold.util.dialog.DialogUtil;
import com.vgaw.scaffold.util.statusbar.StatusBarUtil;
import com.vgaw.scaffold.view.CheckedImageButton;
import com.vgaw.scaffold.view.TitleLayout;
import com.vgaw.scaffolddemo.R;

public class OtherFrag extends BaseFrag {
    private NestedScrollView mOtherRoot;
    private CheckedImageButton mOtherCheckIb;
    private CheckedImageButton mOtherCheckIb1;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.other_frag, container, false);
        StatusBarUtil.addStatusbarHeight(getContext(), view);
        mOtherRoot = view.findViewById(R.id.other_root);

        // toolbar
        TitleLayout otherTitleLayout = view.findViewById(R.id.other_title_layout);
        otherTitleLayout.setBackClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DialogUtil.dismissDialog(getSelf());
            }
        });

        // checkable imagebutton
        mOtherCheckIb = view.findViewById(R.id.other_checkedib);
        mOtherCheckIb.setNormalImageId(android.R.drawable.checkbox_off_background);
        mOtherCheckIb.setCheckedImageId(android.R.drawable.checkbox_on_background);
        mOtherCheckIb.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mOtherCheckIb.setChecked(!mOtherCheckIb.isChecked());
            }
        });

        mOtherCheckIb1 = view.findViewById(R.id.other_checkedib1);
        mOtherCheckIb1.setNormalBkResId(android.R.drawable.checkbox_off_background);
        mOtherCheckIb1.setCheckedBkResId(android.R.drawable.checkbox_on_background);
        mOtherCheckIb1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mOtherCheckIb1.setChecked(!mOtherCheckIb1.isChecked());
            }
        });
        return view;
    }
}

File: app/src/main/java/com/vgaw/scaffolddemo/page/demo/compoment/other/OtherFrag.kt
package com.vgaw.scaffolddemo.page.demo.compoment.other

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.vgaw.scaffold.page.ScaffoldFrag
import com.vgaw.scaffold.util.dialog.DialogUtil
import com.vgaw.scaffold.util.statusbar.StatusBarUtil
import com.vgaw.scaffold.view.CheckedImageButton
import com.vgaw.scaffold.view.TitleLayout
import com.vgaw.scaffolddemo.R

class OtherFrag : ScaffoldFrag() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.other_frag, container, false)
        StatusBarUtil.addStatusbarHeight(context, view)

        // toolbar
        view.findViewById<TitleLayout>(R.id.other_title_layout).setBackClickListener {DialogUtil.dismissDialog(getSelf())}

        // checkable imagebutton
        val otherCheckedib = view.findViewById<CheckedImageButton>(R.id.other_checkedib)
        otherCheckedib.setNormalImageId(android.R.drawable.checkbox_off_background)
        otherCheckedib.setCheckedImageId(android.R.drawable.checkbox_on_background)
        otherCheckedib.setOnClickListener {otherCheckedib.setChecked(!otherCheckedib.isChecked())}

        val otherCheckedib1 = view.findViewById<CheckedImageButton>(R.id.other_checkedib1)
        otherCheckedib1.setNormalBkResId(android.R.drawable.checkbox_off_background)
        otherCheckedib1.setCheckedBkResId(android.R.drawable.checkbox_on_background)
        otherCheckedib1.setOnClickListener {otherCheckedib1.setChecked(!otherCheckedib1.isChecked())}
        return view
    }
}
